### DISCUSSION TITLE
Sector presists in TerminateWait for a long time

### DISCUSSION THREAD

**Discussion by potato1992** [filecoin-project/lotus#7343] [Storage Provider] [answered]
One of my sector reported finalized fail, thus was terminated by me a week ago, but it still persists in TerminateWait status.
When will it disappear?

By the way, I have finally found that the finalize failure is caused by my change to the code, now I have already fixed it. is it possible that I force set the status of this sector to finalize so that it will retry finalize?

**Comment by de-authority** [comment, marked as answer]
Have you ever restart the miner?
I think you may not recover the sector. because the terminal message will send on chain.
but you can try to set the sector to the previous state see what happens.

**Reply by potato1992** [reply]
> Have you ever restart the miner?
> I think you may not recover the sector. because the terminal message will send on chain.
> but you can try to set the sector to the previous state see what happens.

I have restarted the miner. It's ok I cannot recover it, but how long will this sector be wiped out from my sector list?

**Reply by de-authority** [reply]
`lotus-miner sectors remove --really-do-it <sectorNum> `

**Reply by s0nik42** [reply]
yes remove the sector files and restart

**Reply by potato1992** [reply]
thank you.
